### PR TITLE
export in csv

### PR DESCRIPTION
**Pull request by kashishchakarvarti**

### CODE REVIEW

File: api/app/App.js [hunk 1]
```diff
@@ -273,4 +273,33 @@ module.exports = function() {
             console.log(e);
         }
     });
+
+    //Export Table for :appID
+    global.app.post('/backup/:appId/:tableName/exporttable', function(req, res) {
```
Comment (nawazdhandala) [inline api/app/App.js]: Change the API to export/:appId/:tableName

File: api/app/App.js [hunk 2]
```diff
@@ -273,4 +273,33 @@ module.exports = function() {
...
+        console.log("++++ Export Table ++++++");
+        try {
+            var appKey = req.body.key;
+            var appId = req.params.appId;
+            var tableName = req.params.tableName;
+            var exportType = req.body.exportType.toLowerCase();
+
+            global.appService.isMasterKey(appId,appKey).then(function(isMasterKey) {
+                if (isMasterKey) {
+                    global.appService.exportTable(appId,tableName,exportType).then(function(data) {
```
Comment (nawazdhandala) [inline api/app/App.js]: MasterKey should be passed in this function too.

File: api/app/App.js [hunk 3]
```diff
@@ -273,4 +273,33 @@ module.exports = function() {
...
+    global.app.post('/backup/:appId/:tableName/exporttable', function(req, res) {
+        console.log("++++ Export Table ++++++");
+        try {
+            var appKey = req.body.key;
+            var appId = req.params.appId;
+            var tableName = req.params.tableName;
+            var exportType = req.body.exportType.toLowerCase();
+
+            global.appService.isMasterKey(appId,appKey).then(function(isMasterKey) {
+                if (isMasterKey) {
```
Comment (nawazdhandala) [inline api/app/App.js]: Remove this if.

File: api/app/App.js [hunk 4]
```diff
@@ -273,4 +273,33 @@ module.exports = function() {
...
+                    global.appService.exportTable(appId,tableName,exportType).then(function(data) {
+                        res.status(200).send(data);
```
Comment (nawazdhandala) [inline api/app/App.js]: Does this send the file back?

File: services/app.js [hunk 5]
```diff
@@ -965,8 +966,28 @@ module.exports = function() {
                     deferred.resolve({username: username, password: password})
             });
             return deferred.promise;
-        }
+        },
+
+        exportTable : function(appId,tableName,exportType){
```
Comment (nawazdhandala) [inline services/app.js]: Have last param as masterKey

File: services/app.js [hunk 6]
```diff
@@ -965,8 +966,28 @@ module.exports = function() {
...
 
+            var deferred = q.defer();
+
+            var collection = global.mongoClient.db(appId).collection(tableName);
```
Comment (nawazdhandala) [inline services/app.js]: Use find function in custom service.

File: services/app.js [hunk 7]
```diff
@@ -965,8 +966,28 @@ module.exports = function() {
...
+            var findQuery = collection.find({});
+            findQuery.toArray(function(err, tables) {
+                if (err) {
+                    deferred.reject("Error : Failed to retrieve the table.");
+                }
+                if(exportType === 'csv')
```
Comment (nawazdhandala) [inline services/app.js]: Have support for xls and json.

File: services/app.js [hunk 8]
```diff
@@ -965,8 +966,28 @@ module.exports = function() {
...
+                {
+                    jsonexport(tables,function(err, csv){
```
Comment (nawazdhandala) [inline services/app.js]: Convert json objects to string and arrays to string before you export.

**Comment by nawazdhandala** [comment]
Also, there are conflicts too.

**Comment by nawazdhandala** [comment]
Also fix code-climate issues. :)

**Comment by shubhamqweasd** [comment]
re-send a fresh pr , without conflicts

**Pull request closed without merging**